### PR TITLE
uint support + varray lock upgrade

### PR DESCRIPTION
**Pull request by SerGeRybakov**
1. support for numpy `uint`s added
- np.uint,
- np.uint8,
- np.ubyte,
- np.uint16,
- np.uint32,
- np.uintc,
- np.uint64,
- np.uintp,
- np.ushort,
- np.ulonglong,

2. VArray is not locked on data writing
3. Uri refactored
4. Attributes may indicated in schemas as `None` 
3. DekerMemoryError message improved

### CODE REVIEW

File: deker/client.py [hunk 1]
```diff
@@ -214,16 +214,20 @@ def __init__(
         try:
             set_logging_level(loglevel.upper())
             self.__get_plugins()
-            mem_limit = convert_human_memory_to_bytes(memory_limit)
+            total_available_mem = virtual_memory().total + swap_memory().total
+            memory_limit = convert_human_memory_to_bytes(memory_limit)
+            if memory_limit >= total_available_mem or memory_limit <= 0:
```
Comment (matveyvarg) [inline deker/client.py:219]: I find such limitation  unobvious. We do we need to decide for user how much memory he'll be using ? In my opinion, it's better to have it as it was.

File: deker/locks.py [hunk 2]
```diff
@@ -310,18 +316,27 @@ def release(self, e: Optional[Exception] = None) -> None:  # noqa[ARG002]
             Path(f"{lock}:{os.getpid()}{LocksExtensions.varray_lock.value}").unlink(missing_ok=True)
         super().release()
 
+    def acquire(self, path: Optional[Path]) -> None:
+        """VArray shall not be locked itself.
```
Comment (matveyvarg) [inline deker/locks.py:320]: Either:
Varray shall not be locked on itself
or
Varray shall not lock itself

File: poetry.lock
Comment (matveyvarg) [inline poetry.lock:1]: Did you update requirements on purpose ?

File: tests/parameters/schemas_params.py [hunk 3]
```diff
@@ -520,7 +520,7 @@ def _generate_types(
             datetime.now(),
             Scale(0.1, 0.2),
         ]:
-            if exception_types and type(item) in exception_types:
+            if exception_types and type(item) in exception_types or item is None:
```
Comment (matveyvarg) [inline tests/parameters/schemas_params.py:523]: I believe `or item is None` is a bit extra here

File: tests/test_cases/test_tools/test_tools.py [hunk 4]
```diff
@@ -6,12 +6,12 @@
 
 from deker_local_adapters import LocalCollectionAdapter
 
-from deker.tools.array import generate_uid
 from tests.parameters.collection_params import CollectionParams
 
 from deker.collection import Collection
 from deker.errors import DekerInstanceNotExistsError, DekerMemoryError, DekerValidationError
 from deker.tools import check_memory, convert_human_memory_to_bytes
+from deker.tools.array import generate_uid
```
Comment (matveyvarg) [inline tests/test_cases/test_tools/test_tools.py:14]: Doesn't seem like there is any usage

File: deker/client.py (hunk 1 above)
Comment (SerGeRybakov) [reply deker/client.py:219]: User is physically limited to the full memory of machine. If he wants 1000 Tb, but has only 100 Gb, how can it become 1000 Tb?

File: tests/test_cases/test_tools/test_tools.py (hunk 4 above)
Comment (SerGeRybakov) [reply tests/test_cases/test_tools/test_tools.py:14]: Line 227

**Review by matveyvarg — APPROVED** [review]

**Pull request merged**